asciidoc/modules/con_mg_migrating-clusters.adoc

```diff
@@ -0,0 +1,40 @@
+[id="migrating_clusters_{context}"]
+
+= Migrating clusters
+
+It is important to decide the migration strategy for your codebase in a cluster. This is because you cannot add {VertX} {v3x} nodes and {VertX} {v4} nodes together in a single cluster for the following reasons: 
+
+* Cluster manager upgrades - If there are major version upgrades in cluster managers, they may not have support all features for backward compatibility. You will have to check the updates in cluster managers.
+* Subscription data changes - If the information stored in subscription maps or caches is no longer the same in the cluster manager versions. 
+* Transport protocol changes - If some fields in the message transport protocol have changed in the cluster.
+
+If you have a {VertX} cluster for a single application or for some closely related microservices, you can migrate the entire codebase to the new cluster at one time.
+
+However, if you cannot migrate the codebase at one time, use the recommendations in this section to migrate a {VertX} {v3x} codebase to {VertX} {v4}.
+
+== Splitting the cluster
+
+If you have a cluster where different teams have deployed verticles for their applications, you can consider splitting the {VertX} {v3x} cluster into smaller ones. Note that after splitting the cluster, the separated components will not be able to communicate using the clustering features. You can splitthe cluster using the following XXX:
+
+* EventBus request and reply - HTTP/RESTful web services, gRPC
+* EventBus send and publish - Messaging systems such as, Kafka, Postgres `LISTEN` and `NOTIFY`, Redis Pub and Sub
+* Shared Data - Redis, Infinispan
+
+When you split the cluster, each team can move to {VertX} {v4} when they are ready or if it is needed.
+
+== Using {VertX} EventBus Link
+
+If you cannot split your cluster, then use https://github.com/vert-x3/vertx-eventbus-link[Vert.x EventBus Link] to migrate your codebase incrementallyXXX.
+
+Vert.x EventBus Link is a tool that connects a {VertX} {v3x} clustered EventBus to a {VertX} {v4} clustered EventBus.
+
+WARNING: The migration of shared data API, that is, maps, counters and locks is not supported.
+
+The tool creates an `EventBusLink` object that implements the `EventBus` interface. An instance of `EventBusLink` is created on one node of each cluster. The instance is created by providing a set of addresses and its behavior depends on the message paradigmXXX:
+
+* _fire and forget_ and _request/reply_ - The message is sent (XXX instaed of forwarded) to the remote cluster which delegates (XXX what?) to the {VertX} EventBus.
+* _publish_ - The message is sent (XXX instaed of forwarded) to the remote cluster and to the {VertX} EventBus.
+
+The {VertX} EventBus Link creates a WebSocket server to receive messages and uses a WebSocket client to send them.
+
+See the sections link:https://github.com/vert-x3/vertx-eventbus-linkdocumentation for more details.
```

Comment:
There are references to the different sections missing, I think

Reply:
Updated the links.